runtime/vm/resolvesupport.cpp

```diff
@@ -2297,17 +2297,18 @@ resolveInvokeDynamic(J9VMThread *vmThread, J9ConstantPool *ramCP, UDATA callSite
 			result = vmThread->currentException;
 		} else if (result == NULL) {
 			setCurrentExceptionUTF(vmThread, J9VMCONSTANTPOOL_JAVALANGNULLPOINTEREXCEPTION, NULL);
-		} else { /* (result != NULL) */
-			/* Only write the value in if it is not null */
-			J9MemoryManagerFunctions *gcFuncs = vmThread->javaVM->memoryManagerFunctions;
-			J9Class *j9class = J9_CLASS_FROM_CP(ramCP);
+			result = vmThread->currentException;
+		}
 
-			/* Ensure that result array elements are written before the array reference is stored */
-			VM_AtomicSupport::writeBarrier();
-			if (0 == gcFuncs->j9gc_objaccess_staticCompareAndSwapObject(vmThread, j9class, callSite, NULL, result)) {
-				/* Another thread beat this thread to updating the call site, ensure both threads return the same method handle. */
-				result = *callSite;
-			}
+		/* Only write the value in if it is not null */
```

Comment:
Delete the comment as this doesn't apply any more - we're writing something, the array or exception, provided no one else has beaten us to it.

```suggestion
```

Reply:
updated.